pkg/api/aim/query/query.go

```diff
@@ -581,6 +612,57 @@ func (pq *parsedQuery) parseName(node *ast.Name) (any, error) {
 					}
 				},
 			), nil
+		case "re":
+			return callable(
+				func(args []ast.Expr) (any, error) {
+					operation, ok := args[0].(*ast.Str)
+					if !ok {
+						return nil, errors.New("re operation type has to be ast.Str")
+					}
+					switch operation.S {
+					case RegexpOperationMatch:
+						fallthrough
+					case RegexpOperationSearch:
+						// subtract 1, because first argument is an operation argument.
+						if len(args)-1 != 2 {
+							return nil, errors.New("re.match function support exactly 2 arguments")
+						}
+
+						regexp, ok := args[1].(*ast.Str)
+						if !ok {
+							return nil, errors.New("first argument type for re.match function has to be ast.Str")
+						}
+						attribute, ok := args[2].(*ast.Attribute)
+						if !ok {
+							return nil, errors.New("second argument type for re.match function has to be ast.Attribute")
```

Comment:
second argument should be parsed with `parseNode` and the result should be checked to be a `clause.Column` — this will make sure that things like `re.match("regex", run.tags["mlflow.source.name"])` work and should make the code much simpler

Reply:
done.